### PR TITLE
#1390 - Remove gender from bcsc integration for Students and Supporting Users

### PR DESCRIPTION
**Pull request by dheepak-aot**
### Student
- [x] Remove gender from BCSC login integration (including mappers in keycloak)
- [x] Move gender field out of the area that says "go to BCSC to change"
image
image

- [x] Change Student readonly Gender field to a dropdown with the values
- Male (male)
- Female (female)
- Unknown (unknown)
- Diverse (diverse)

### Supporting Users
- [x] Remove Gender field from supporting users form and database table

image
image

### Form definition local deployment
- [x] Validated local form deployment after updating form definitions
image

### CODE REVIEW

File: sources/packages/backend/apps/api/src/route-controllers/student-account-applications/models/student-account-application.dto.ts [hunk 1]
```diff
@@ -37,6 +37,4 @@ export class StudentAccountApplicationApprovalAPIInDTO extends CreateStudentAPII
   email: string;
   @Allow()
   dateOfBirth: string;
-  @Allow()
-  gender: string;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student-account-applications/models/student-account-application.dto.ts:41]: Why was this removed from the Ministry approval? The Ministry should be able to change any information and create the student. This information was already coming from the form, not from BCSC. Can you please elaborate on why was it removed or what I am missing here?

File: sources/packages/backend/apps/api/src/services/student/student.service.ts [hunk 2]
```diff
@@ -336,23 +336,23 @@ export class StudentService extends RecordDataModelService<Student> {
   /**
    * Updates the student contact information.
    * @param studentId student to be updated.
-   * @param contact contact information to be updated.
+   * @param studentInfo student information to be updated.
    * @param auditUserId user who is making the changes.
    * @returns updated student.
    */
   async updateStudentContactByStudentId(
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/student/student.service.ts:343]: The method should be renamed also. Gender is not part of the contact information.

File: sources/packages/forms/src/form-definitions/studentprofile.json [hunk 3]
```diff
@@ -642,84 +642,97 @@
...
+                      "inputFormat": "plain",
+                      "protected": false,
                       "dbIndex": false,
+                      "case": "",
+                      "truncateMultipleSpaces": false,
+                      "encrypted": false,
+                      "redrawOn": "",
+                      "clearOnHide": true,
                       "customDefaultValue": "",
+                      "calculateValue": "value = data.email;",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/studentprofile.json:679]: Not part of the PR, I noticed the `calculateValue` as it is here in other form.io definitions, unless I am missing something it is only needed if the `data` property is different than the input API name. If they are the same this `calculateValue` is useless.

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, just a minor comment.

**Review by andrepestana-aot — APPROVED** [review]
LGTM!

File: sources/packages/backend/apps/api/src/route-controllers/student/student.students.controller.ts [hunk 4]
```diff
@@ -331,7 +331,7 @@ export class StudentStudentsController extends BaseController {
         "Not able to update a student due to an invalid request.",
       );
     }
-    await this.studentService.updateStudentContactByStudentId(
+    await this.studentService.updateStudentInfoStudentId(
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/route-controllers/student/student.students.controller.ts:334]: I guess that there is a "by" missing `updateStudentInfoByStudentId`. Either way `updateStudentInfo` would be ok IMO also.

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍
